Sources/GoodNetworking/DataRequestExtensions.swift

```diff
@@ -33,6 +33,39 @@ public extension DataRequest {
         preprocessor: DataPreprocessor = DecodableResponseSerializer<T>.defaultDataPreprocessor,
         emptyResponseCodes: Set<Int> = DecodableResponseSerializer<T>.defaultEmptyResponseCodes,
         emptyResponseMethods: Set<HTTPMethod> = DecodableResponseSerializer<T>.defaultEmptyRequestMethods
+    ) -> AnyPublisher<T, AFError> {
+        let serializer = DecodableResponseSerializer<T>(
+            dataPreprocessor: preprocessor,
+            decoder: JSONDecoder(),
+            emptyResponseCodes: emptyResponseCodes,
+            emptyRequestMethods: emptyResponseMethods
+        )
+        return self.validate()
+            .publishResponse(using: serializer, on: queue)
+            .value()
+    }
+    
+    /// Creates a `DataResponsePublisher` for this instance and uses a ``DecodableResponseSerializer`` to serialize the
+    /// response.
+    ///
+    /// - Parameters:
+    ///   - type:                `Decodable` type to which to decode response `Data`. Inferred from the context by default.
+    ///   - queue:               `DispatchQueue` on which the `DataResponse` will be published. `.main` by default.
+    ///   - preprocessor:        `DataPreprocessor` which filters the `Data` before serialization. `PassthroughPreprocessor()`
+    ///                          by default.
+    ///   - decoder:             `DataDecoder` instance used to decode response `Data`. `JSONDecoder()` by default.
+    ///   - emptyResponseCodes:  `Set<Int>` of HTTP status codes for which empty responses are allowed. `[204, 205]` by
+    ///                          default.
+    ///   - emptyRequestMethods: `Set<HTTPMethod>` of `HTTPMethod`s for which empty responses are allowed, regardless of
+    ///                          status code. `[.head]` by default.
+    ///
+    /// - Returns:               The `DataResponsePublisher`.
+    func goodify<T: Decodable & WithCustomDecoder>(
```

Comment:
I would add decoder in all goodify functions as a parameter with default value. 

cc: @plajdo